docs/zkapps/o1js/basic-concepts.md

```diff
@@ -127,25 +126,106 @@ You can think assertions in o1js similar to `require()` function in Solidity.
+<!-- prettier-ignore -->
+```ts
+console.log(Field.ORDER);
 ```
 
-However, note that these methods are _not_ provable, meaning you cannot use them in your final zkApp as a part of the ZKP. Use them only to debug your code.
+When you perform an operation inside a finite field, the result always stays in the range. This is achieved through [modular arithmetic](https://en.wikipedia.org/wiki/Modular_arithmetic). This behaviour may be unfamiliar to you if you have not worked with ZKPs before. Let us explain more with some examples.
+
+In the addition or multiplication operations where the result is bigger than the order of the finite field, you receive the modulus of the result with the prime order of the field.
 
 <!-- prettier-ignore -->
 ```ts
-const x = Field(45);
-const y = Field(x.toString()); // You should NOT write this
+Provable.log(Field(Field.ORDER - 1) + Field(2)); // This will print 1
 ```
 
-## Other Types and Methods
+If you try creating a `Field` from a negative number you will receive the modular additive inverse of the number. For instance `(1 + (Field.ORDER - 1)) % Field.ORDER` is 0, so the additive inverse of 1 is `Field.ORDER - 1` in this field.
 
-In addition to the `Field` type, o1js gives you a variety structures to help you build a powerful zkApp. You can learn more about them in the next chapter, [Common Types and Functions](./common-types-and-functions.md).
+<!-- prettier-ignore -->
+```ts
+Provable.log(Field(-1)); // This will print `Field.ORDER - 1`
+```
+
+Similarly, if your substraction results in a negative number you will get the modular additive inverse.
+
+For division, if the result is not an integer, you receive the [modular multiplicative inverse](https://en.wikipedia.org/wiki/Modular_multiplicative_inverse) of the division: If `(x * y) % p` is `z` (where `p` is the order of the finite field), then the modular multiplicative inverse of the division `z / x` equals `y`. Note that `x`, `y`, and `z` are all integers.
+
+<!-- prettier-ignore -->
+```ts
+Provable.log(Field(3).div(Field(5))); // This will print a very big integer
+```
+
+Because of this behaviour, it is usually not very useful to use the final result of an uneven division. Instead, you can prove the result with multiplication.
+
+<!-- prettier-ignore -->
+```ts
+// We want x, an argument of the smart contract, to be equal to 7 / 3
+x.mul(Field(3)).assertEquals(Field(7));
+```
+
+If you want to have integer division instead, you can use `UInt32` or `UInt64` types (see [Common Types and Functions](./common-types-and-functions.md)).
+
+## Provability and Provable Code in o1js
+
+We have stated many times above that all the code you write in o1js must be provable, but what is provability?
+
+A provable code can be thought as a code that o1js compiler can convert into a ZKP. Just because you write something with o1js, it does not become automatically provable. You must follow all the rules above and always use provable types in your code.
```

Comment:
> o1js compiler

let's not use the term "compiler", it's a confusion already that people think we compile TS to circuits